src/vm/i386/gmsx86.cpp

```diff
@@ -1295,10 +1295,10 @@ void LazyMachState::unwindLazyState(LazyMachState* baseState,
     ctx.Esi = lazyState->_esi = baseState->_esi;
     ctx.Ebx = lazyState->_ebx = baseState->_ebx;
 
-    nonVolRegPtrs.Edi = &(lazyState->_edi);
-    nonVolRegPtrs.Esi = &(lazyState->_esi);
-    nonVolRegPtrs.Ebx = &(lazyState->_ebx);
-    nonVolRegPtrs.Ebp = &(lazyState->_ebp);
+    nonVolRegPtrs.Edi = &(baseState->_edi);
```

Comment:
@janvorli  I tried to implement the fixup approach as you suggested. I found that the fixup code would be different from other architectures due to the difference in ``LazyMachState`` structure. 

To be specific, all the other architectures use  the following layout for ``LazyMachState``:
```
struct LazyMachState  {
   TADDR CalleeSavedRegisters[...];
   TADDR CalleeSavedRegisterPointers[...];
}
```
But, x86 interleaves callee saved registeres with callee saved register pointers:
```
    PTR_TADDR _pEdi;
    TADDR     _edi;
    PTR_TADDR _pEsi;
    TADDR     _esi;
    PTR_TADDR _pEbx;
    TADDR     _ebx;
    PTR_TADDR _pEbp;
    TADDR     _ebp;
```

Should I change ``LazyMachState``?

While reading the code, ``HelperMethodFrame::InsureInit`` is the only function that ``LazyMachState::unwindLazyState``, and both ``LazyMachState::unwindLazyState``  and ``setLazyStateFromUnwind`` are implemented in gmscpu.h (and gmsx86.cpp).

I'm not sure, but it seems that this implementation is related with architecture rather than EH model. Could you share your opinion?

@jkotas Could you please take a look?